src/Symfony/Component/Validator/Tests/Constraints/TwigValidatorTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Tests\Constraints;
+
+use Symfony\Component\Validator\Constraints\Twig;
+use Symfony\Component\Validator\Constraints\TwigValidator;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+
+/**
+ * @author Mokhtar Tlili <tlili.mokhtar@gmail.com>
+ */
+class TwigValidatorTest extends ConstraintValidatorTestCase
+{
+    protected function createValidator(): TwigValidator
+    {
+        return new TwigValidator();
+    }
+
+    /**
+     * @dataProvider getValidValues
+     */
+    public function testTwigIsValid($value)
+    {
+        $this->validator->validate($value, new Twig());
+
+        $this->assertNoViolation();
+    }
+
+    /**
+     * @dataProvider getInvalidValues
+     */
+    public function testInvalidValues($value, $message, $line)
+    {
+        $constraint = new Twig('myMessageTest');
+
+        $this->validator->validate($value, $constraint);
+
+        $this->buildViolation('myMessageTest')
+            ->setParameter('{{ error }}', $message)
+            ->setParameter('{{ line }}', $line)
+            ->setCode(Twig::INVALID_TWIG_ERROR)
+            ->assertRaised();
+    }
+
+    public static function getValidValues()
+    {
+        return [
+            ['Hello {{ name }}'],
+            ['{% if condition %}Yes{% else %}No{% endif %}'],
+            ['{# Comment #}'],
+            ['Hello {{ "world" | upper }}'],
+            ['{% for i in 1..3 %}Item {{ i }}{% endfor %}'],
+        ];
+    }
+
+    public static function getInvalidValues()
+    {
+        return [
+            // Invalid syntax example (missing end tag)
+            ['{% if condition %}Oops', 'Unexpected end of template at line 1.', 1],
+            // Another syntax error example (unclosed variable)
+            ['Hello {{ name', 'Unexpected token "end of template" ("end of print statement" expected) at line 1.', 1],
+            // Unknown filter error
+            ['Hello {{ name | unknown_filter }}', 'Unknown "unknown_filter" filter at line 1.', 1],
```

Comment:
Here, with the default environment, all the following would be invalid template, even if they are on most of the Symfony docs page.

```twig
{% form_theme form resources %}
```

```twig
{{ name | trans }}  
```

```twig
{% trans_default_domain domain %}
```

```twig
{{ object|serialize(format = 'json', context = []) }}
```

```twig
{{ text|humanize }}
```

```twig
{{ name | trans }}  
```

So ... what does  "Twig valid"  mean ?

Reply:
Resolved by injecting the environment, as suggested by @derrabus https://github.com/symfony/symfony/pull/58805#discussion_r1833913662